ethereumj-core/src/main/java/org/ethereum/Start.java

```diff
@@ -17,45 +17,79 @@
  */
 package org.ethereum;
 
-import org.apache.commons.lang3.StringUtils;
 import org.ethereum.cli.CLIInterface;
 import org.ethereum.config.SystemProperties;
 import org.ethereum.facade.Ethereum;
 import org.ethereum.facade.EthereumFactory;
 import org.ethereum.mine.Ethash;
 
 import java.io.IOException;
-import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static java.lang.Long.parseLong;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
 
 /**
  * @author Roman Mandeleil
  * @since 14.11.2014
  */
 public class Start {
 
-    public static void main(String args[]) throws IOException, URISyntaxException {
+    public static void main(String args[]) {
         CLIInterface.call(args);
 
         final SystemProperties config = SystemProperties.getDefault();
-        final boolean actionBlocksLoader = !config.blocksLoader().isEmpty();
-        final boolean actionGenerateDag = !StringUtils.isEmpty(System.getProperty("ethash.blockNumber"));
 
-        if (actionBlocksLoader || actionGenerateDag) {
-            config.setSyncEnabled(false);
-            config.setDiscoveryEnabled(false);
-        }
+        getEthashBlockNumber().ifPresent(blockNumber -> {
```

Comment:
Could you detach this block of code to private method with Javadoc?

Reply:
Why do we need add more methods to this pretty simple logic? I guess the main method is readable and straightforward enough.

Reply:
I'd like to see it with only line `createEthereum` 😃